### PR TITLE
Fixed cChunk::m_Entities corruption upon world travel

### LINKED ISSUE #2724: Crash on multiworld travel
I was in the Nether. I executed `/portal world`

```
[70153d24f3fd4a92|15:28:10] Assertion failed: std::find(m_Entities.begin(), m_Entities.end(), a_Entity) == m_Entities.end(), file /home/cuberite/cuberite/src/Chunk.cpp, line 1944
./Cuberite_debug(_Z15PrintStackTracev+0x23)[0xb0c901]
./Cuberite_debug(_ZN6cChunk9AddEntityEP7cEntity+0xce)[0xa0e5f4]
./Cuberite_debug(_ZN6cChunk20MoveEntityToNewChunkEP7cEntity+0xfe)[0xa0af96]
./Cuberite_debug(_ZN6cChunk4TickENSt6chrono8durationIlSt5ratioILl1ELl1000EEEE+0x3e6)[0xa0ac24]
./Cuberite_debug(_ZN9cChunkMap11cChunkLayer4TickENSt6chrono8durationIlSt5ratioILl1ELl1000EEEE+0x86)[0xa23ecc]
./Cuberite_debug(_ZN9cChunkMap4TickENSt6chrono8durationIlSt5ratioILl1ELl1000EEEE+0x58)[0xa2347a]
./Cuberite_debug(_ZN6cWorld4TickENSt6chrono8durationIlSt5ratioILl1ELl1000EEEES4_+0x3f6)[0xad8a12]
./Cuberite_debug(_ZN6cWorld11cTickThread7ExecuteEv+0xad)[0xacf425]
./Cuberite_debug(_ZN9cIsThread9DoExecuteEv+0x2f)[0xb01d8d]
./Cuberite_debug(_ZNKSt7_Mem_fnIM9cIsThreadFvvEEclIIEvEEvPS0_DpOT_+0x65)[0xb0338d]
./Cuberite_debug(_ZNSt12_Bind_simpleIFSt7_Mem_fnIM9cIsThreadFvvEEPS1_EE9_M_invokeIILm0EEEEvSt12_Index_tupleIIXspT_EEE+0x43)[0xb032dd]
./Cuberite_debug(_ZNSt12_Bind_simpleIFSt7_Mem_fnIM9cIsThreadFvvEEPS1_EEclEv+0x1d)[0xb031d3]
./Cuberite_debug(_ZNSt6thread5_ImplISt12_Bind_simpleIFSt7_Mem_fnIM9cIsThreadFvvEEPS3_EEE6_M_runEv+0x1c)[0xb03150]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0xb6970)[0x7f3acdeb1970]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x80a4)[0x7f3ace10e0a4]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f3acd62104d]
Cuberite_debug: /home/cuberite/cuberite/src/Chunk.cpp:1944: void cChunk::AddEntity(cEntity*): Assertion `0' failed.

```

### PR DESCRIPTION
**Pull request by LogicParrot**
Fixes #2724
possibly fixes or completely resolves other [world travel issues](https://github.com/cuberite/cuberite/milestones/Stable%20world%20travel) as well.

### CODE REVIEW

**Comment by LogicParrot** [comment]
This should be merged into testing.

File: src/Entities/Entity.cpp [hunk 1]
```diff
@@ -1499,8 +1498,18 @@ bool cEntity::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
...
+		return true;
+	}))
+	{
+		LOGD("Entity Teleportation failed! Didn't find the source chunk!\n");
+		return false;
+	}
+
 	// Remove all links to the old world
-	SetWorldTravellingFrom(GetWorld());  // cChunk::Tick() handles entity removal
+	// SetWorldTravellingFrom(GetWorld());  // cChunk::Tick() handles entity removal
```
Comment (LogicParrot) [inline src/Entities/Entity.cpp]: Delete this line.

**Comment by Seadragon91** [comment]
Tested this works fine. Great work :)

**Comment by LogicParrot** [comment]
Thanks :)
This invalidates iterators when traveling through a nether portal, which eventually sigfaults. This explains why it was done the old way in the first place. I'll need to update this or take a different approach.

**Comment by LogicParrot** [comment]
Replaced with #2941, closing

**Pull request closed without merging**